clap_complete/src/dynamic/completer.rs

```diff
@@ -704,3 +723,110 @@ impl CompletionCandidate {
         self.visible
     }
 }
+
+/// User-provided completion candidates for an argument.
+///
+/// This is useful when predefined value hints are not enough.
+pub trait CustomCompleter: core::fmt::Debug + Send + Sync {
+    /// All potential candidates for an argument.
+    ///
+    /// See [`CompletionCandidate`] for more information.
+    fn completions(&self) -> Vec<CompletionCandidate>;
+}
+
+/// Implement `CustomCompleter` for a closure
+///
+/// # Example
```

Comment:
We've documented this on `ArgValueCompleter`, I don't think we need to provide detailed examples on the impl